### DISCUSSION TITLE
SSL in winlogbeat - logstash

### DISCUSSION THREAD

**Discussion by jozb09** [Security-Onion-Solutions/securityonion#9427] [General]
I managed to follow the steps on documentation, https://docs.securityonion.net/en/2.3/beats.html

Now, I need to know what should I do on agent side? 
<img width="244" alt="image" src="https://user-images.githubusercontent.com/18210409/208398704-0b77c9c3-a77d-4129-a7b9-c8316ed2e6cc.png">
 Should I uncomment those SSL lines?

And how can I test to see if my SSL config works and if logs are shipping in an encrypted format. ??

Thanks!

**Comment by weslambert** [comment]
You can copy the `myca.crt/mybeats.crt` to your endpoint and use it for the appropriate values.

You can test to see if data is encrypted by using something like tcpdump to sniff the traffic hitting the interface/port of the Security Onion node.

**Reply by alexandruhera** [reply]
This is the kind of answer I was looking for as well, but , where are these certs stored? Moreover, since the elasticsearch now wants authentication, is there way for us to define the user/password, or API key? It's more difficult to edit the elastic yml file in the container without breaking anything.

**Reply by alexandruhera** [reply]
Never mind, I found some in /etc/pki, I will try them out.

**Comment by jozb09** [comment]
@weslambert I used internal PKI which I have cer file with private key

**Reply by alexandruhera** [reply]
Did you copy those files to the target winlogbeat machine?

**Reply by jozb09** [reply]
@alexandruhera Yes I did, and I specify the path like the below screenshot, 

<img width="377" alt="image" src="https://user-images.githubusercontent.com/18210409/209657031-b0d8cbb2-28f7-43f8-bf11-3046091c3706.png">

**Reply by alexandruhera** [reply]
Thanks a lot, I haven't realized that. Also, what about the authentication for the elastic output? Do you know where these are set, in which config file? :)

**Reply by jozb09** [reply]
under /opt/so/saltstack/local/salt/logstash/pipelines/config/so/0009_input_beats.conf

**Reply by jozb09** [reply]
Can you share what should be configured to that file?

**Reply by alexandruhera** [reply]
Absolutely, I will come back on this, but now it appears that my containers can't boot..
